esx_service/utils/vm_listener.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+VM change listener (started as a part of vmdkops service).
+It monitors VM poweroff events and detaches the DVS managed
+volumes from the VM and updates the status in KV
+'''
+
+import logging
+import os
+import os.path
+import atexit
+import time
+import threadutils
+import log_config
+import vmdk_utils
+import vmdk_ops
+
+from pyVmomi import VmomiSupport, vim, vmodl
+
+VM_POWERSTATE = 'runtime.powerState'
+POWERSTATE_POWEROFF = 'poweredOff'
+HOSTD_RECONNECT_INTERVAL = 20
+
+def start_vm_changelistener():
+    """
+    Listen to power state changes of VMs running on current host
+    """
+    si = vmdk_ops.get_si()
+
+    while not si:
```

Comment:
I am always scared of infinite loops. Does this loop break if you stop/kill service?